library/ui/src/main/java/com/google/android/exoplayer2/ui/TrackSelectionView.java

```diff
@@ -367,6 +380,48 @@ private void onTrackViewClicked(View view) {
     }
   }
 
+  private TrackGroupArray initSortedTrackGroups(TrackGroupArray trackGroups) {
+    TrackGroupArray trackGroupArray = trackGroups;
+    if(comparator != null) {
+      TrackGroupArray trackGroupsArray = mappedTrackInfo.getTrackGroups(rendererIndex);
+      for (int groupIndex = 0; groupIndex < trackGroupsArray.length; groupIndex++) {
+        TrackGroup group = trackGroupsArray.get(groupIndex);
+        Format[] listFormats = new Format[group.length];
+        for (int formatIndex = 0; formatIndex < group.length; formatIndex++) {
+          listFormats[formatIndex] = group.getFormat(formatIndex);
+        }
+        Arrays.sort(listFormats, comparator);
+        trackGroupArray = new TrackGroupArray(new TrackGroup(listFormats));
+      }
+    }
+    return trackGroupArray;
+  }
+
+  /**
+   * The correspondence between trackGroup and sortedTrackGroup indexes.
+   * initial array (only quality for this example) : [480,720,256,1080]
+   * asc sorted array (only quality for this example) : [256,480,720,1080]
+   * initial array index for 480 is 0, but for sorted array index is 1.
+   * Initial array index is @param trackIndex, and the @return result is sorted array index
+   * @param groupIndex which TrackGroup you want to browse into
+   * @param trackIndex which index of the initial array
+   * @return index of the sorted array that correspond to the same element in the initial array
+   */
+  private int getSortedIndexFromInitialTrackGroup(int groupIndex, int trackIndex) {
+    int sortedTrackIndex = trackIndex;
+    if(sortedTrackGroups != trackGroups) {
+      Format selectedFormat = sortedTrackGroups.get(rendererIndex).getFormat(trackIndex);
+      int trackHash = selectedFormat.hashCode();
+      for (int formatIndex = 0; formatIndex < trackGroups.get(groupIndex).length; formatIndex++) {
+        if(trackGroups.get(groupIndex).getFormat(formatIndex).hashCode() == trackHash) {
```

Comment:
You can't rely on hash codes to determine equality. More generally though, I don't think it needs to be this complicated. There's a `TrackGroup.indexOf(Format)` method that you can use. I think you can probably replace the whole `for` loop with a call to this method.

Reply:
You're totally right, I've changed for `indexOf` and it's working the same without the `for` loop